server/src/test/java/io/crate/metadata/functions/SignatureBinderTest.java

```diff
@@ -615,12 +642,31 @@ public void produces(BoundVariables expected) {
             assertThat(actual).isEqualTo(expected);
         }
 
+        public void hasReturnType(DataType<?> expected) {
+            BoundSignature actual = bind();
```

Comment:
consider creating a custom assertj assertion (e.g. `SettingsAssert`) for BoundSignature, but given the method `bind()` which is needed, maybe doesn't make sense.